gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -249,6 +253,10 @@ public void apply(JobListener jobListener, JobContext jobContext) throws Excepti
             return;
           }
 
+          WriterInitializer writerInitializer = WriterInitializerFactory.newInstace(jobState, workUnits.get());
+          closer.register(writerInitializer);
```

Comment:
You can combine the above two lines as `closer.register` returns the object it registers